### PR TITLE
Implement max fps limitation and improve firmware stability

### PR DESCRIPTION
**Pull request by GergelySzabolcs**
This pr contains:
- max fps limitation for streams
- fix for occasional crashes

### CODE REVIEW

**Comment by itsderek23** [comment]
I need to head out shortly so I can't do a full review, but `test.py` runs for me w/o immediately freezing! So that's a great first step. 

When I return I'm going to test out `fps_limit`.

**Comment by itsderek23** [comment]
# Error w/depth_sipp, left, right (possibly more) streams

If I try the following command (with and without resetting prior):

```
python3 test.py -co '{"streams": ["depth_sipp"]}'
```

I see a repeat of `Device get data failed: 7` errors and the stream is not displayed. [Full log](https://gist.github.com/itsderek23/d918e81f62790d83f3fe1da9afec05f7).

The default streams (metaout, previewout) don't trigger this. I'm seeing this with at least the following stream names:

* left
* depth_sipp
* right

**Comment by itsderek23** [comment]
It seems like if `previewout` is not in the list of `streams`, the error occurs.

**Comment by itsderek23** [comment]
The `max_fps` setting does appear to work for me.

File: test.py [hunk 1]
```diff
@@ -70,10 +70,10 @@ def parse_args():
 # Do not modify the default values in the config Dict below directly. Instead, use the `-co` argument when running this script.
 config = {
     # Possible streams:
-    # ['left', 'right','previewout', 'metaout', 'depth_sipp']
+    # ['left', 'right','previewout', 'metaout', 'depth_sipp', 'disparity', 'depth_color_h']
     # If "left" is used, it must be in the first position.
     # To test depth use:
-    # ['metaout', 'previewout', 'depth_sipp']
+    # 'streams': [{'name': 'depth_sipp', "max_fps": 12.0}, {'name': 'previewout', "max_fps": 12.0}, ],
```
[the remark this answers is not in the document]
Comment (itsderek23) [reply test.py]: What happens if the streams list contains values in both formats? For example:

```
    'streams': [{'name': 'depth_sipp', "max_fps": 12.0}, 'previewout'],

```

**Comment by itsderek23** [comment]
Great @chris-piekarski. Thx for letting us know. What streams are you using? I didn't have luck using `depth_sipp` by itself.

**Comment by chris-piekarski** [comment]
Agreed @itsderek23. I gave you a +1 on your comment regarding that above. I had to set previewout to fps 1 for depth_sipp to work.

File: test.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (GergelySzabolcs) [reply test.py]: depth_sipp will be limited to 12 fps, previewout won't be limited.

File: test.py [hunk 2]
```diff
@@ -70,10 +70,10 @@ def parse_args():
...
     'streams': ['metaout', 'previewout'],
```
[the remark this answers is not in the document]
Comment (GergelySzabolcs) [reply test.py]: If max_fps is not provided there will be no limitation, the output fps will depend on how the cameras are configured (30 fps for rgb, and stereo)

**Comment by itsderek23** [comment]
Great! I can now display single streams. I made one fix: https://github.com/luxonis/depthai-python-extras/pull/41/commits/dd4381dec4deea3ddf2d269e8311dad005c85df9.

Thanks @GergelySzabolcs - I think this does it.

**Pull request merged**